packages/zero-cache/src/workers/syncer.ts

```diff
@@ -115,25 +116,46 @@ export class Syncer implements SingletonService {
 
     let decodedToken: JWTPayload | undefined;
     if (auth) {
-      try {
-        decodedToken = await verifyToken(this.#authConfig, auth, {
-          subject: userID,
-        });
-        this.#lc.debug?.(
-          `Received auth token ${auth} for clientID ${clientID}, decoded: ${JSON.stringify(decodedToken)}`,
-        );
-      } catch (e) {
-        sendError(
-          this.#lc,
-          ws,
-          {
-            kind: ErrorKind.AuthInvalidated,
-            message: `Failed to decode auth token: ${String(e)}`,
-          },
-          e,
+      const tokenOptions = tokenConfigOptions(this.#config.auth);
+
+      const hasPushOrMutate =
+        this.#config?.push?.url !== undefined ||
+        this.#config?.mutate?.url !== undefined;
+      const hasGetQueries = this.#config?.getQueries?.url !== undefined;
+
+      // must either have one of the token options set or have custom mutations & queries enabled
+      assert(
```

Comment:
It was already like this in prev code but in the future don't use assertions for this kind of user-level error. use normal `throw new Error...`. `assert` is for checking invariants - a failed assertion means that a Rocicorp dev screwed up, not the user.

Reply:
cc @tantaman